### ISSUE TITLE
NullPointerException in handlerPackage

### ISSUE THREAD

**Issue by ejuniorasas** [springfox/springfox#2080]
Hello after I updated to 2.7.0 I start receiving a NullPointerException
from the class:
springfox.documentation.builders.RequestHandlerSelectors
when call handlerPackage method (new method it doesn't exist on 2.6.1)
and when input is "com.sun.proxy.$Proxy312"
input.getPackage() - returns null

  ```
private static Function<Class<?>, Boolean> handlerPackage(final String basePackage) {
    return new Function<Class<?>, Boolean>() {
      @Override
      public Boolean apply(Class<?> input) {
        return input.getPackage().getName().startsWith(basePackage);
      }
    };
  }
```

Maybe I forgot to configure something.

Thanks!

**Comment by dilipkrish** [comment]
Sounds like a cglib/aop issue where your proxy class doesnt belong to any package.

**Comment by ejuniorasas** [comment]
Ok thanks, 
It doesn't happen at 2.6.1 did I forget to configure something? That I saw this class was changed "springfox.documentation.builders.RequestHandlerSelectors" 
before we have:

  ```
/**
   * Predicate that matches RequestHandler with given base package name for the class of the handler method.
   * This predicate includes all request handlers matching the provided basePackage
   *
   * @param basePackage - base package of the classes
   * @return this
   */
  public static Predicate<RequestHandler> basePackage(final String basePackage) {
    return new Predicate<RequestHandler>() {
      @Override
      public boolean apply(RequestHandler input) {
        return declaringClass(input).getPackage().getName().startsWith(basePackage);
      }
    };
  }

  private static Class<?> declaringClass(RequestHandler input) {
    return input.declaringClass();
  }
```
And now we have more tow private methods, that were this null point appear.
```
  private static Function<Class<?>, Boolean> annotationPresent(final Class<? extends Annotation> annotation) {
    return new Function<Class<?>, Boolean>() {
      @Override
      public Boolean apply(Class<?> input) {
        return input.isAnnotationPresent(annotation);
      }
    };
  }


  private static Function<Class<?>, Boolean> handlerPackage(final String basePackage) {
    return new Function<Class<?>, Boolean>() {
      @Override
      public Boolean apply(Class<?> input) {
        return input.getPackage().getName().startsWith(basePackage);
      }
    };
  }

  /**
   * Predicate that matches RequestHandler with given base package name for the class of the handler method.
   * This predicate includes all request handlers matching the provided basePackage
   *
   * @param basePackage - base package of the classes
   * @return this
   */
  public static Predicate<RequestHandler> basePackage(final String basePackage) {
    return new Predicate<RequestHandler>() {
      @Override
      public boolean apply(RequestHandler input) {
        return declaringClass(input).transform(handlerPackage(basePackage)).or(true);
      }
    };
  }

  private static Optional<? extends Class<?>> declaringClass(RequestHandler input) {
    return Optional.fromNullable(input.declaringClass());
  }
```

Sorry about my insistence, but I am try to use the springforx new version with Springboot latest (1.5.8.RELEASE)

**Comment by dilipkrish** [comment]
Oh yeah. I'll try to release a fix in a week or so. Thanks for reporting again tho'

**Comment by dilipkrish** [comment]
Fixed as part of #1876
